### PR TITLE
Fix sync_bn grad maker

### PR DESCRIPTION
**Pull request by sneaxiy**
`BatchNormGradMaker` is defined twice in `batch_norm_op.cc` and `sync_batch_norm_op.cc` respectively, which would make gcc link error. This PR fixes it.

### CODE REVIEW

**Review by gongweibao — DISMISSED** [review]
LGTM

**Review by gongweibao — APPROVED** [review]
LGTM
